Dalamud/Game/Gui/NamePlate/NamePlateGui.cs

```diff
@@ -0,0 +1,302 @@
+﻿using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+using Dalamud.Game.Addon.Lifecycle;
+using Dalamud.Game.Addon.Lifecycle.AddonArgTypes;
+using Dalamud.Game.ClientState.Objects;
+using Dalamud.IoC;
+using Dalamud.IoC.Internal;
+using Dalamud.Plugin.Services;
+
+using FFXIVClientStructs.FFXIV.Client.UI;
+
+namespace Dalamud.Game.Gui.NamePlate;
+
+/// <summary>
+/// Class used to modify the data used when rendering nameplates.
+/// </summary>
+[ServiceManager.EarlyLoadedService]
+internal sealed class NamePlateGui : IInternalDisposableService, INamePlateGui
+{
+    /// <summary>
+    /// The index for the number array used by the NamePlate addon.
+    /// </summary>
+    public const int NumberArrayIndex = 5;
+
+    /// <summary>
+    /// The index for the string array used by the NamePlate addon.
+    /// </summary>
+    public const int StringArrayIndex = 4;
+
+    /// <summary>
+    /// The index for of the FullUpdate entry in the NamePlate number array.
+    /// </summary>
+    internal const int NumberArrayFullUpdateIndex = 4;
+
+    /// <summary>
+    /// An empty null-terminated string pointer allocated in unmanaged memory, used to tag removed fields.
+    /// </summary>
+    internal static readonly nint EmptyStringPointer = CreateEmptyStringPointer();
```

Comment:
This will leak, is that intentional?

Reply:
It's intentional. This will live for the scope of Dalamud and I think it's worth a byte basically. If we freed this we could potentially end up with bad data in the string array.